xyz.slothlife.Jogger.json

```diff
@@ -0,0 +1,104 @@
+{
+  "app-id": "xyz.slothlife.Jogger",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "45",
+  "sdk": "org.gnome.Sdk",
+  "sdk-extensions": [
+    "org.freedesktop.Sdk.Extension.rust-stable",
+    "org.freedesktop.Sdk.Extension.llvm16"
+  ],
+  "command": "jogger",
+  "finish-args": [
+    "--share=network",
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    "--env=RUST_BACKTRACE=1"
+  ],
+  "build-options": {
+    "append-path": "/usr/lib/sdk/rust-stable/bin:/usr/lib/sdk/llvm16/bin",
+    "env": {
+      "CARGO_REGISTRIES_CRATES_IO_PROTOCOL": "sparse",
+      "RUST_BACKTRACE": "1",
+      "RUST_LOG": "jogger=debug"
+    }
+  },
+  "cleanup": [
+    "/include",
+    "/lib/pkgconfig",
+    "/man",
+    "/share/doc",
+    "/share/gtk-doc",
+    "/share/man",
+    "/share/pkgconfig",
+    "*.la",
+    "*.a"
+  ],
+  "modules": [
+    {
+      "name": "libshumate",
+      "buildsystem": "meson",
+      "config-opts": [
+        "-Dvapi=false",
+        "-Dgtk_doc=false"
+      ],
+      "builddir": true,
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://download.gnome.org/sources/libshumate/1.1/libshumate-1.1.0.tar.xz",
+          "sha256": "b3b7c4c2d3002f37b951c9ca99b09275d0aa82265c69f53374ce12faefd3b14e"
+        }
+      ]
+    },
+    {
+      "name": "blueprint-compiler",
+      "buildsystem": "meson",
+      "cleanup": ["*"],
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://gitlab.gnome.org/jwestman/blueprint-compiler",
+          "tag": "v0.10.0",
+          "commit": "2a39a16391122af2f3d812e478c1c1398c98b972"
+        }
+      ]
+    },
+    {
+      "name": "espeak-ng",
+      "buildsystem": "autotools",
+      "no-parallel-make": true,
+      "build-options": {
+        "ldflags": "-Wl,--allow-multiple-definition"
+      },
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/espeak-ng/espeak-ng.git",
+          "tag": "1.51",
+          "commit": "2e9a5fccbb0095e87b2769e9249ea1f821918ecd"
+        }
+      ]
+    },
+    {
+      "name": "jogger",
+      "builddir": true,
+      "buildsystem": "meson",
+      "build-commands": [
+        "cargo --offline fetch --manifest-path Cargo.toml",
+        "cargo --offline build --release"
```

Comment:
these are supposed to be run by meson. I'm not even sure they'll be run.

But you have to set `CARGO_HOME`

Reply:
Presumably I'd need some sort of switch in my meson build then, for it to know if it's being run from Flathub or not?